wavesexchange_apis/src/api_clients/node.rs

```diff
@@ -269,4 +278,92 @@ pub mod dto {
         String { value: String },
         // todo rest of them
     }
+
+    impl DataEntryResponse {
+        #[inline]
+        pub fn key(&self) -> &str {
+            match self {
+                DataEntryResponse::String { key, .. }
+                | DataEntryResponse::Integer { key, .. }
+                | DataEntryResponse::Boolean { key, .. }
+                | DataEntryResponse::Binary { key, .. } => key.as_str(),
+            }
+        }
+
+        #[inline]
+        pub fn into_value(self) -> DataEntryValue {
+            match self {
+                DataEntryResponse::String { value, .. } => DataEntryValue::String(value),
+                DataEntryResponse::Integer { value, .. } => DataEntryValue::Integer(value),
+                DataEntryResponse::Boolean { value, .. } => DataEntryValue::Boolean(value),
+                DataEntryResponse::Binary { value, .. } => DataEntryValue::Binary(value),
+            }
+        }
+    }
+
+    impl Value {
+        #[inline]
+        pub fn try_as_array(&self) -> Result<&[Value], TypeError> {
+            match self {
+                Value::Array { value } => Ok(value.as_slice()),
+                _ => Err(self.type_error("Array")),
+            }
+        }
+
+        #[inline]
+        pub fn try_into_tuple_2(self) -> Result<Result<(Value, Value), TupleError>, TypeError> {
+            match self {
+                Value::Tuple { value } => Ok(Self::hash_map_into_tuple_2(value)),
+                _ => Err(self.type_error("Tuple")),
+            }
+        }
```

Comment:
Nested `Result<Result<>>` here is a dirty temporary solution.
Error handling needs careful design across the whole crate.
Need a more general error type which will also include some context information.